List WMI available | Linux local Fix

lapolis opened the pull request:
"Security Updates for Microsoft Office Products C2R (August 2023)" with plugin ID 179614 is part of plugin family "Windows" instead of "Windows : Microsoft Bulletins".

"Funny" thing is that "Security Updates for Microsoft Office Products C2R (July 2022)" is correctly places in plugin family "Windows : Microsoft Bulletins".

This would also make the filtering at line 582/585 to let that issue ending up in the "Missing Linux Patches" tab.

lapolis committed:
Update nessusToExcel.py

"Security Updates for Microsoft Office Products C2R (August 2023)" with plugin ID 179614 is part of plugin family "Windows" instead of "Windows : Microsoft Bulletins".

"Funny" thing is that "Security Updates for Microsoft Office Products C2R (July 2022)" is correctly places in plugin family "Windows : Microsoft Bulletins".

spyr0-sec commented:
Thanks again for this. I'm going to raise a case with Tenable to confirm if its been correctly tagged and leave this PR open in the meantime. Hope that's alright

samwmarsh commented:
@lapolis I've raised a case with tenable, however it appears looking on their plugin search that C2R are always marked with the family "Windows", apart from July 2022. 
![image](https://github.com/spyr0-sec/helper-scripts/assets/74866556/f9f08ae9-2c9f-49bf-88b4-72e0f551d7e8)

lapolis commented:
Oh right. So we might have to come up with a different solution. Filtering by partial name maybe?

```
... or ("Security Updates for Microsoft " in plugin_name):
```

samwmarsh commented:
Update from Tenable when asking

1. Why are they Windows and not Windows: Microsoft Bulletin
2. Can we just get July 2022 updated to "Windows" to bring all the C2R inline (or move them _all_ to Microsoft Bulletin

Upon investigation, your assessment appears correct - I will check with the plugins team on why these are considered to belong to the Windows rather than Windows: Bulletin family (or contrariwise) and get back to you with my findings.

I'll keep this thread updated as/when they reply.

lapolis commented:
Ah well, moving all those to Microsoft Bulletin would make sense. Good call contacting them.

lapolis committed:
Update nessusToExcel.py

Added module to extract all WMI Available hosts to quickly grab a list of auth scanned hosts.

lapolis commented:
In the meantime I did a new commit with a new module that I've been using recently. Just mentioning in case you were wondering what that second commit was :) Sometime I do a mixed scan against entire subnets so having a clean list of hosts that were cred patch scanned is handy. I did that searching for plugin "WMI Available". Let me know if you want me to change something.

Note that the second commit do not include the first one as I reverted the changes in my code just in case Tenable moves everything to _Microsoft Bulletin_.

lapolis committed:
Parsing Linux outdated software

This was not being picked up:

```
  - Installed package : linux-image-4.4.0-31-generic_4.4.0-31.50
    Fixed package     : linux-image-4.4.0-<ANY>-generic_4.4.0-169.198

  - Installed package : linux-image-generic_4.4.0.31.33
    Fixed package     : linux-image-generic_4.4.0.169.177
```

Pluign Info:

```
ID:131014
Version:1.11
Type:local
Family:Ubuntu Local Security Checks
```

lapolis commented:
In the meantime I also added a fix for the Linux module.

This was not being picked up:

```
  - Installed package : linux-image-4.4.0-31-generic_4.4.0-31.50
    Fixed package     : linux-image-4.4.0-<ANY>-generic_4.4.0-169.198

  - Installed package : linux-image-generic_4.4.0.31.33
    Fixed package     : linux-image-generic_4.4.0.169.177
```

Pluign Info:

```
ID:131014
Version:1.11
Type:local
Family:Ubuntu Local Security Checks
```

spyr0-sec commented:
LGTM, thanks again! I'll let you know if Tenable get back to us

samwmarsh commented:
They’ve moved it to the Windows family and closed the case

lapolis commented:
Nice. Well done flagging that to them!
